File: sources/Astolfy B1.0/igbt/astolfy/module/player/InvMove.java
package igbt.astolfy.module.player;

import org.lwjgl.input.Keyboard;

import igbt.astolfy.events.Event;
import igbt.astolfy.events.listeners.EventUpdate;
import igbt.astolfy.module.ModuleBase;
import net.minecraft.client.gui.GuiChat;

public class InvMove extends ModuleBase {

	public InvMove() {
		super("InvMove", Keyboard.KEY_NONE, Category.PLAYER);
	}
	
	public void onEvent(Event e) {
		if(e instanceof EventUpdate) {
	        if (mc.currentScreen != null && !(mc.currentScreen instanceof GuiChat)) {
				mc.gameSettings.keyBindForward.pressed = org.lwjgl.input.Keyboard.isKeyDown(mc.gameSettings.keyBindForward.getKeyCode());
				mc.gameSettings.keyBindLeft.pressed = org.lwjgl.input.Keyboard.isKeyDown(mc.gameSettings.keyBindLeft.getKeyCode());
				mc.gameSettings.keyBindBack.pressed = org.lwjgl.input.Keyboard.isKeyDown(mc.gameSettings.keyBindBack.getKeyCode());
				mc.gameSettings.keyBindRight.pressed = org.lwjgl.input.Keyboard.isKeyDown(mc.gameSettings.keyBindRight.getKeyCode());
				mc.gameSettings.keyBindJump.pressed = org.lwjgl.input.Keyboard.isKeyDown(mc.gameSettings.keyBindJump.getKeyCode());
				mc.gameSettings.keyBindSprint.pressed = org.lwjgl.input.Keyboard.isKeyDown(mc.gameSettings.keyBindSprint.getKeyCode());
	        }
		}
	}

}

File: sources/Leaf devEnv/com/leafclient/leaf/mod/movement/InvMove.kt
package com.leafclient.leaf.mod.movement

import com.leafclient.leaf.event.game.input.KeyboardTickEvent
import com.leafclient.leaf.extension.isKeyPressed
import com.leafclient.leaf.management.mod.ToggleableMod
import com.leafclient.leaf.management.mod.category.Category
import fr.shyrogan.publisher4k.subscription.Listener
import fr.shyrogan.publisher4k.subscription.Subscribe
import net.minecraft.client.gui.inventory.GuiContainer
import org.lwjgl.input.Keyboard

class InvMove: ToggleableMod("InvMove", Category.MOVEMENT) {

    private val keys = arrayOf(
        mc.gameSettings.keyBindForward, mc.gameSettings.keyBindBack,
        mc.gameSettings.keyBindLeft, mc.gameSettings.keyBindRight,
        mc.gameSettings.keyBindJump
    )

    /**
     * Modifies the pressed statue of keys in inventory to allow movement input
     */
    @Subscribe
    val onKeyboardTick = Listener<KeyboardTickEvent> { e ->
        if(mc.currentScreen is GuiContainer) {
            keys
                .filter { it.keyCode >= 0 }
                .forEach {
                    it.isKeyPressed = Keyboard.isKeyDown(it.keyCode)
                }
        } else if(mc.currentScreen == null) {
            keys
                .filter { !Keyboard.isKeyDown(it.keyCode) }
                .forEach { it.isKeyPressed =  false }
        }
    }

    init {
        isRunning = true
    }

}